cylc/flow/task_events_mgr.py

```diff
@@ -284,6 +286,28 @@ def process_events(self, schd_ctx):
             elif ctx.ctx_type == self.HANDLER_JOB_LOGS_RETRIEVE:
                 self._process_job_logs_retrieval(schd_ctx, ctx, id_keys)
 
+    def _spawn_children(self, itask, output):
+        """Spawn children of output, or all children on succeed/fail"""
+        try:
+            children = itask.children[output]
+        except KeyError:
+            # No children depend on this ouput
+            children = []
+
+        finished = False
+        if output in [TASK_OUTPUT_SUCCEEDED, TASK_OUTPUT_FAILED]:
+            # Spawn all children
+            childrens = itask.children.values()
+            children = list(chain(*childrens))
+            if not (output == TASK_OUTPUT_FAILED
+                    and not itask.failure_handled):
```

Comment:
Double negative!

```suggestion
            if output != TASK_OUTPUT_FAILED and itask.failure_handled):
```

I feel this should be `==` rather than `!=` though:

```
            if output == TASK_OUTPUT_FAILED and itask.failure_handled):
```

Reply:
Damn it, I don't not like double negatives.

Reply:
Going for this:
```python
if output in [TASK_OUTPUT_SUCCEEDED, TASK_OUTPUT_FAILED]:
    >snip<
    # Only succeeded and handled-failed tasks are considered finished.
    finished = TASK_OUTPUT_SUCCEEDED or itask.failure_handled 
```